src/Discovery/DiscoveryDetails/DataDownloadList/DataDownloadList.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Collapse, List, Button } from 'antd';
+import { DownloadOutlined } from '@ant-design/icons';
+import CheckThatDataHasTitles from './Utils/CheckThatDataHasTitles';
+import DataDownloadListItem from './Utils/DataDownloadListItem';
+
+const { Panel } = Collapse;
+
+const DataDownloadList = ({ sourceFieldData }: any) => {
+  const data = sourceFieldData[0].map((obj) => (
+    {
+      title: obj.title,
+      description: obj.description,
+    }
+  ),
+  );
+  if (CheckThatDataHasTitles(data) === false) return null;
```

Comment:
if `title` is missing then we shoud try using `file_name`, and if both are missing then we can ignore it

Reply:
Updated data object used for the DataDownload to use the title from the sourceField object _or_ the file_name value if the title is not there. Otherwise will evaluate to null and component will not render. 

```  
const data = sourceFieldData[0].map((obj) => ({
    title: obj.title || obj.file_name,
    description: obj.description,
}));

```

Reply:
Wait I think logic of `CheckThatDataHasTitles()` is still a bit different, looks like it is enforcing that every entries in `data` should have either `tilte` or `file_name`. But what we really need is that it would be okay if some entries don't have them, and then we just ignore those and display the rest in the list.
This can be done by something like `const data = sourceFieldData[0].filter(...).map(...)`, and when we are filtering, use `console.debug` to leave a debug message for those entries who neither has a `title` nor a `file_name`

Reply:
Got it - I removed the CheckThatDataHasTitles and added logic in src/Discovery/DiscoveryDetails/DataDownloadList/Utils/ProcessData.ts. Added a test to validate it console debugs if the object does not have a title or file_name. Updated DataDownloadList line 10 to not render the component if there are zero DataDownloadListItems available. Updated DataDownloadList.test.tsx to test for this as well. 

I think that in QA all the entries will now render a DataDownloadList. For example: 
<img width="733" alt="image" src="https://github.com/uc-cdis/data-portal/assets/113449836/60f56538-7dea-47e1-b1b4-a4f9b1471db0">